Client/core/CChat.cpp

```diff
@@ -771,17 +776,26 @@ bool CChat::CharacterKeyHandler(CGUIKeyEventArgs KeyboardArgs)
     return true;
 }
 
-void CChat::SetVisible(bool bVisible)
+void CChat::SetVisible(bool bVisible, bool bInputBlocked)
 {
     m_bVisible = bVisible;
-    // If hiding chat, also reset chat input line
+
     if (!m_bVisible)
-        SetInputVisible(false);
+    {
+        m_bInputBlocked = bInputBlocked;
+
+        if (m_bInputBlocked)
+            SetInputVisible(false);
+    }
+    else
+    {
+        m_bInputBlocked = false;
+    }
```

Comment:
What if you want to show the chat but block input?

If you do `showChat(true, true)`, the chat is shown but you're still able to type. I don't see any reason why we *shouldn't* allow that.